staging/src/k8s.io/legacy-cloud-providers/azure/azure_config.go

```diff
@@ -0,0 +1,122 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package azure
+
+import (
+	"fmt"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/klog"
+	"sigs.k8s.io/yaml"
+)
+
+const (
+	secretNamespace      = "kube-system"
+	secretCloudConfigKey = "cloud-config"
+)
+
+// The configure type for Azure cloud provider secret. Supported values are:
+// * all            : configure applied for components (kubelet and controller-manager). This is the default value.
+// * node           : configure applied for nodes (kubelet).
+// * control-plane  : configure applied for control plane components (controller-manager).
+//
+// For different configure types, the secret name would also be different:
+// * all            : secret name would be azure-cloud-provider.
+// * node           : secret name would azure-cloud-provider-node.
+// * control-plane  : secret name would be azure-cloud-provider-control-plane.
+type secretConfigureType string
+
+const (
+	secretConfigureAll          secretConfigureType = "all"
+	secretConfigureNode         secretConfigureType = "node"
+	secretConfigureControlPlane secretConfigureType = "control-plane"
+)
+
+// The override type for Azure cloud provider secret. Supported values are:
+// * no   : The values from secret won't override any configures from local cloud-config file.
+// * must : The values from secret would override all configures from local cloud-config file.
+// * can  : The values from secret would override only configurations that are explicitly set in the secret. This is the default value.
+type secretOverrideType string
+
+const (
+	secretOverrideTypeNo   secretOverrideType = "no"
+	secretOverrideTypeMust secretOverrideType = "must"
```

Comment:
I find `secretOverrideTypeMust` is only used in test code, looks like `secretOverrideTypeMust` is a useless option in your code, only for doc

Reply:
hmm, it should be checked somewhere. Would add that and ensure only allowed options are accepted.

Reply:
Fixed now.